vktrace/src/vktrace_extensions/vktracevulkan/vkreplay/vkreplay_vkreplay.cpp

```diff
@@ -1566,8 +1566,22 @@ VkResult vkReplay::manually_replay_vkWaitForFences(packet_vkWaitForFences* pPack
     {
         (*(pFence + i)) = m_objMapper.remap_fences((*(pPacket->pFences + i)));
     }
-    replayResult = m_vkFuncs.real_vkWaitForFences(remappedDevice, pPacket->fenceCount, pFence, pPacket->waitAll, pPacket->timeout);
-    VKTRACE_DELETE(pFence);
+	if (pPacket->result == VK_SUCCESS)
+	{
+		replayResult = m_vkFuncs.real_vkWaitForFences(remappedDevice, pPacket->fenceCount, pFence, pPacket->waitAll, UINT64_MAX);// mean as long as possible
+	}
```

Comment:
We had a few cases where the timeout from the application wasn't long enough during replay, and replay would stop because of different return return values (received TIMEOUT when SUCCESS was expected). I guess we could continually loop and keep the app's original timeout. At least then the additional wait is inside the vkreplay code instead of waiting inside the driver. I'll review this and see if we can simplify these conditionals.